app/src/components/ProfileEdit/QuestionField.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from "react"
+import { useParams, useHistory } from "react-router"
+import { useHostHomeData } from "../../data/data-context"
+import { ProgressPlugin } from "webpack"
+import { Question } from "../../models/Question"
+import styled from "styled-components"
+import { FormControl, FormControlLabel, FormGroup, RadioGroup, Radio, Checkbox, TextField } from '@material-ui/core';
+
+const QuestionContainer = styled.div`
+  min-height: 244px;
+`
+
+interface Props {
+  question: Question,
+  setAnswer: (id: string, value: any) => void
+};
+
+const QuestionField = (props: Props) => {
+  const { question } = props;
+
+  function handleChange(e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) {
+    const questionId = props.question.id;
+    if (e.target.type === 'checkbox') {
+      if ('checked' in e.target) {
+        props.question.answer[e.target.value] = e.target.checked;
+      }
+      props.setAnswer(questionId, props.question.answer);
+    } else {
+      props.setAnswer(questionId, e.target.value);
+    }
+  }
+
+  return (
+    <QuestionContainer>
+      <h2>{question.question}</h2>
+      <div>
+        {
+          question.type === 'radio' && 
+          <FormControl component="fieldset">
+            <RadioGroup aria-label={question.question} key={question.id} name={question.id} value={props.question.answer} onChange={handleChange}>
+              {
+                question.options && question.options.map((option: any) => 
+                  <FormControlLabel key={option.value} value={option.value} control={<Radio />} label={option.label} />
+                )
+              }
+            </RadioGroup>
+          </FormControl>
+        }
+        {
+          question.type === 'checkbox' &&
+          <FormControl component="fieldset">
```

Comment:
we should build the MUI components out as generic components so we can reuse them through the app. i think we can thinking about this in a separate task / PR but wanted to mention it.

Reply:
Sure, it's always good to not repeat oneself. NRO = not repeat oneself